crates/services/p2p/src/codecs.rs

```diff
@@ -1,19 +1,24 @@
+pub mod bounded;
 pub mod postcard;
+pub mod unbounded;
 
-use crate::{
-    gossipsub::messages::{
-        GossipTopicTag,
-        GossipsubBroadcastRequest,
-        GossipsubMessage,
-    },
-    request_response::messages::{
-        RequestMessage,
-        V2ResponseMessage,
-    },
-};
+use crate::gossipsub::messages::GossipTopicTag;
 use libp2p::request_response;
+use serde::{
+    Deserialize,
+    Serialize,
+};
 use std::io;
 
+trait DataFormatCodec {
+    type Error;
+    fn deserialize<'a, R: Deserialize<'a>>(
+        encoded_data: &'a [u8],
+    ) -> Result<R, Self::Error>;
+
+    fn serialize<D: Serialize>(data: &D) -> Result<Vec<u8>, Self::Error>;
```

Comment:
This could also be generalized by taking a `impl std::io::Write` input which it writes to. Then a convenience function `serialize_to_vec` could be provided with this signature.

Reply:
This has been changed to reflect what's been done in the Storage crate. We now have an `Encoder` trait that is used as the type where the result is stored, and this is implemented for `Cow<'_, [u8]>`. I could in theory change this to have the input where the serialized data is written passed into the function, but this would create me trouble when I will try to merge the traits in the storage trait and this PR together (see new issue https://github.com/FuelLabs/fuel-core/issues/2403)